src/openloco/windows/options.cpp

```diff
@@ -629,6 +631,19 @@ namespace openloco::ui::options
             draw_string_494B3F(*dpi, w->x + scale_widget.left + 1, w->y + scale_widget.top + 1, colour::black, string_ids::scale_formatted, &scale);
         }
 
+        static void applyScreenModeRestrictions(window* w)
+        {
+            if (config::get_new().display.mode != config::screen_mode::fullscreen)
+                w->disabled_widgets = (1 << display::widx::display_resolution) | (1 << display::widx::display_resolution_btn);
+
+#if !(defined(__APPLE__) && defined(__MACH__))
```

Comment:
What is the reasoning behind this?

Reply:
macOS is a little particular when it comes to fullscreen resolution. Switching modes through SDL2 would not move OpenLoco to its own space, which has some unexpected side effects. 

Therefore, when we implemented the Options window, @marijnvdwerf suggested we disable mode switching entirely on macOS, leaving the switching up up to OS controls. You can still use a borderless fullscreen window, just through the native (green) fullscreen control in the window's title bar.

This is not new code, by the way, I just refactored it so it doesn't occur twice.